l2geth/core/vm/evm.go

```diff
@@ -323,6 +326,15 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 		}
 	}
 
+	// We need to be able to show a status of 1 ("no error") for successful L1 => L2 messages.
+	// The current way we figure out the success of the transaction (by parsing the
+	// returned data) would require an upgrade to the contracts that we likely won't make for a
+	// while. As a result, we'll use this mechanism where we set IsOkL1ToL2Message = true if
+	// we detect an L1 => L2 message that didn't revert. Initially we just set the value to false.
+	if evm.depth == 0 {
```

Comment:
The zero value for a boolean is false so this is just being explicit

Reply:
I didn't know if geth ever reuses the same `evm` object to do multiple executions, I felt it was safer to reset this for each new call. I can delete if you think this is unnecessary.

Reply:
I agree that having it better, its good to be explicit. Would be surprised if an evm is used more than once